lib/realtime/tenants/check.ex

```diff
@@ -0,0 +1,130 @@
+defmodule Realtime.Tenants.Check do
+  use GenServer
+
+  require Logger
+
+  alias Realtime.Helpers
+
+  def connection_status(tenant_id) do
+    case get_status(tenant_id) do
+      :undefined ->
+        :ok
+        node = Realtime.Nodes.get_node_for_tenant_id(tenant_id)
```

Comment:
This isn't going to work because `get_node_for_tenant_id` ends up calling `Tenants.get_tenant_by_external_id`.

Anywhere in the join path you need to use `Tenants.Cache.get_tenant_by_external_id` otherwise we hit our db too much when we get a flood of reconnects.

Reply:
Will change 👍